src/Controls/src/Core/TitleBar/TitleBar.cs

```diff
@@ -0,0 +1,301 @@
+﻿using Microsoft.Maui.Graphics;
+
+namespace Microsoft.Maui.Controls
+{
+	public partial class TitleBar : ContentView, ITitleBar
+	{
+		public static readonly BindableProperty IconProperty = BindableProperty.Create(nameof(Icon), typeof(ImageSource),
+			typeof(TitleBar), null, propertyChanged: OnIconChanged);
+
+		public static readonly BindableProperty LeadingContentProperty = BindableProperty.Create(nameof(LeadingContent), typeof(IView),
+			typeof(TitleBar), null, propertyChanged: OnLeadingContentChanged);
+
+		public static new readonly BindableProperty ContentProperty = BindableProperty.Create(nameof(Content), typeof(IView),
+			typeof(TitleBar), null, propertyChanged: OnContentChanged);
+
+		public static readonly BindableProperty TrailingContentProperty = BindableProperty.Create(nameof(TrailingContent), typeof(IView),
+			typeof(TitleBar), null, propertyChanged: OnTrailingContentChanged);
+
+		public static readonly BindableProperty TitleProperty = BindableProperty.Create(nameof(Title), typeof(string),
+			typeof(TitleBar), null, propertyChanged: OnTitleChanged);
+
+		public static readonly BindableProperty SubtitleProperty = BindableProperty.Create(nameof(Subtitle), typeof(string),
+			typeof(TitleBar), null, propertyChanged: OnSubtitleChanged);
+
+		public static readonly BindableProperty ForegroundColorProperty = BindableProperty.Create(nameof(ForegroundColor),
+			typeof(Color), typeof(TitleBar), propertyChanged: OnForegroundChanged);
+
+		public static readonly BindableProperty InactiveBackgroundColorProperty = BindableProperty.Create(nameof(InactiveBackgroundColor),
+			typeof(Color), typeof(TitleBar), null);
+
+		public static readonly BindableProperty InactiveForegroundColorProperty = BindableProperty.Create(nameof(InactiveForegroundColor),
+			typeof(Color), typeof(TitleBar), null);
+
+		public ImageSource Icon
+		{
+			get { return (ImageSource)GetValue(IconProperty); }
+			set { SetValue(IconProperty, value); }
+		}
+		
+		public IView? LeadingContent
```

Comment:
I think for these it might be better to just use VisualElement or something, but then to satisfy the interface add an explicit property:

```cs
IView? ITitleBar.LeadingContent => LeadingContent;
```

This way if the user is doing `TitleBar.LeadingContent.` the intellisense is Controls intellisense.